### DISCUSSION TITLE
Usage statistics

### DISCUSSION THREAD

**Discussion by burtaverde** [qdm12/gluetun#818] [Q&A]
Hello, i was wondering if it was possible to get some usage stats such as data, speed and connectivity (drops, online time, server connected to). The vpn providers can't do it since they don't log that stuff, but it still would be useful to compare my transmission download statistics to the tunnelled data. Would confirm that nothing went outside (and i may be a statistics nerd)

**Comment by qdm12** [comment]
Repeating myself a bit from the other discussion;

- bandwidth could be measured once at connection establishment time. Or periodically. But that also depends on what server you are testing against so it's a bit of a tough topic.
- connectivity is tested for every 5s and the vpn is auto-restarted if it drops, so that's already taken care of. Although we could show when it drops as spikes in a time graph.
- Nothing can go outside due to the firewall. It can only lose connectivity (see point above)

See #279 for more information and feel free to subscribe to it to know when there is some progress.

**Comment by burtaverde** [comment]
ohh my bad i didn't see that one. It seems i got a bit too excited about it working and forgot i had already asked for some stuff. I do like that it's being tested, just a little echo to a text file or notification could be easy to implement and help have some traceability after letting it run for a while. For bandwidth i imagine it's not too difficult to check data being currently transferred? 
-I do really like what you have already, it sure made the setup easy. Perhaps i just haven't read enough about it or maybe it doesn't quite indicate its power, especially the firewall.
